kotlin-insight-client/kotlin-insight-client-test-base/src/main/kotlin/com/linkedplanet/kotlininsightclient/CompanyTestOperatorManualImpl.kt

```diff
@@ -0,0 +1,107 @@
+import com.linkedplanet.kotlininsightclient.api.interfaces.GenericInsightObjectOperator
+import com.linkedplanet.kotlininsightclient.api.interfaces.InsightObjectOperator
+import com.linkedplanet.kotlininsightclient.api.model.InsightAttribute.Companion.toReference
+import com.linkedplanet.kotlininsightclient.api.model.InsightAttribute.Companion.toValue
+import com.linkedplanet.kotlininsightclient.api.model.InsightObject
+import com.linkedplanet.kotlininsightclient.api.model.InsightObjectId
+import com.linkedplanet.kotlininsightclient.api.model.getSingleReferenceValue
+import com.linkedplanet.kotlininsightclient.api.model.getStringValue
+
+class CompanyTestOperatorManualImpl(
+    private val insightObjectOperator: InsightObjectOperator,
+    private val countryTestOperatorManualImpl: CountryTestOperatorManualImpl
+    ) : GenericInsightObjectOperator<Company>{
+
+    private val objectTypeId = InsightObjectType.Company.id
+    private val countryRef = TestAttributes.CompanyCountry.attributeId
+    private val name = TestAttributes.CompanyName.attributeId
+
+    private suspend fun toDomain(insightObject: InsightObject) = Company(
+        name = insightObject.getStringValue(name)!!,
+        country = countryTestOperatorManualImpl.getById(insightObject.getSingleReferenceValue(countryRef)!!.objectId).orNull()!!,
+    )
+
+    override suspend fun create(domainObject: Company): Either<InsightClientError, Company> = either {
+        insightObjectOperator.createObject(
+            objectTypeId,
+            name toValue domainObject.name,
+            countryRef toReference domainObject.country?.name?.let { countryName ->
+                insightObjectOperator.getObjectByName(
+                    countryTestOperatorManualImpl.objectTypeId,
+                    countryName,
+                    ::identity
+                ).bind()?.id
+            },
+            toDomain = ::toDomain
+        ).bind()
+    }
+
+    override suspend fun update(domainObject: Company): Either<InsightClientError, Company> = either {
+        val objectByName = insightObjectOperator.getObjectByName(objectTypeId, domainObject.name, ::identity).bind()
+        if (objectByName == null){
+            create(domainObject).bind()
+        } else {
+            val udpatedObject = insightObjectOperator.updateObject(
+                objectByName,
+                name toValue domainObject.name,
+                countryRef toValue domainObject.name,
+            ).bind()
+            toDomain(udpatedObject)
+        }
+    }
+
+    override suspend fun delete(domainObject: Company): Either<InsightClientError, Unit> = either {
+        val objectByName = insightObjectOperator.getObjectByName(objectTypeId, domainObject.name, ::identity).bind()
+            ?: return@either
+        insightObjectOperator.deleteObject(objectByName.id).bind()
+    }
+
+    override suspend fun getByName(name: String): Either<InsightClientError, Company?> =
```

Comment:
It should be possible to override the ::identity calls in the super class to prevent overloading all methods